### PR TITLE
uefi-capsule: Do not modify the BootOrder variable by default

### PR DESCRIPTION
**Pull request by hughsie**
This seems entirely redundant if BootNext is set. If nobody squeaks in a year or so then we can then remove the code that sets BootOrder.

See https://github.com/fwupd/fwupd/discussions/6014

Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [X] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

**Review by superm1 — COMMENTED** [review]
Don't we also want a conffile line for this for people to touch if they're broken? Generating a quirk is a bit heavy for most people.

**Comment by hughsie** [comment]
@vathpela tells me that some old Lenovo system firmwares wouldn't boot from a BootNext that's not in BootOrder. I think adding a config value for this is probably sensible, and if we do break anything still supported by Lenovo we can add a per-vendor HwId quirk like the boot-order one.

**Comment by superm1** [comment]
> @vathpela tells me that some old Lenovo system firmwares wouldn't boot from a BootNext that's not in BootOrder. I think adding a config value for this is probably sensible, and if we do break anything still supported by Lenovo we can add a per-vendor HwId quirk like the boot-order one.

Come to think of it I remember a similar behavior at Dell too on some stuff.

Maybe the right answer is just delete from boot order the next boot to be safe.

**Comment by vathpela** [comment]
I think we're probably better off just making sure it's way down in the order (and when invoked accidentally through the order nothing surprising happens) than doing writes on a regular basis. In theory we should be able to overwrite a variable with the same size without too much overhead, but in practice who knows. If it's not the same size, it's definitely going to claim new flash blocks, which means the "every ~500 installs SetVariable() fails until a GC cycle on the next reboot happens" problem will become more frequent. Which isn't toooo bad, I think the frequency of the error is something like 32k/avg var size/2^n/m where n=number of things doing writes for their boot variables and m=number of variables changed (i.e. 2). But still, more often.

(I am in no way sure it was Lenovo...)

**Comment by hughsie** [comment]
@superm1 should we merge this or drop this in your opinion?

**Comment by superm1** [comment]
I don't feel comfortable changing the default for everything without affirmations from various vendors it's okay.

How about merging code that allows going this way and we ask vendors to test with a flag in place that adjusts it. When we have enough positive comments flip it.

**Comment by hughsie** [comment]
Could we enable it if `#ifdef SUPPORTED_BUILD` is set? or is that too magic?

**Comment by superm1** [comment]
> Could we enable it if `#ifdef SUPPORTED_BUILD` is set? or is that too magic?

I think that's too magical.  I think if we want vendors to test this way to eventually drop it we need them to do it on binary builds in the distros.  So I think conffile is the right way to go.

Could even go as far as sending the value up to LVFS reports so there are enough positive reports from enough vendors then we flip the default.

File: plugins/uefi-capsule/uefi-capsule.quirk [hunk 1]
```diff
@@ -10,9 +10,21 @@ Flags = no-ux-capsule
...
+[85d38fda-fc0e-5c6f-808f-076984ae7978]
+Flags = modify-bootorder
+
+# ASUSTeK COMPUTER INC.
+[65605ed1-09e2-57aa-bd0f-a26c1d35433f]
+Flags = modify-bootorder
+
+# HP
+[93f84748-c854-5d6b-b78a-13c2361e0758]
+Flags = modify-bootorder
```
Comment (superm1) [inline plugins/uefi-capsule/uefi-capsule.quirk:27]: What about framework

Comment (hughsie) [reply plugins/uefi-capsule/uefi-capsule.quirk:31]: Indeed, added.

**Review by superm1 — APPROVED** [review]

**Pull request merged**